core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -255,17 +346,46 @@ private void rebuildIndex() throws Exception {
       }
    }
 
+   private void migrateFromV12_0() {
+      String cacheName = ctx.getCache().getName();
+      PERSISTENCE.startMigratingPersistenceData(cacheName);
+      File newFile = new File(file.getParentFile(), cacheName + "_new.dat");
+      if (newFile.exists()) {
+         newFile.delete();
+      }
+      try (FileChannel newChannel = SecurityActions.openFileChannel(newFile)) {
+         // Write latest Magic and copy old content
+         newChannel.truncate(0);
+         newChannel.write(ByteBuffer.wrap(MAGIC_LATEST), 0);
+         channel.position(MAGIC_12_0.length);
+         newChannel.transferFrom(channel, MAGIC_12_0.length, channel.size());
```

Comment:
Should this be `newChannel.transferFrom(channel, MAGIC_12_0.length, channel.size() - MAGIC_12_0.length);` ? Maybe it just ignores if count is larger than the amount of available bytes?

Ah yeah it does `long max = Math.min(count, src.size() - pos);`

Reply:
> Maybe it just ignores if count is larger than the amount of available bytes?

Yeah the docs say the following:

```qoute
Fewer than the requested number of bytes will be transferred if the source channel has fewer than count bytes remaining, or if the source channel is non-blocking and has fewer than count bytes immediately available in its input buffer.
```